extensions/binding/org.eclipse.smarthome.binding.mqttbroker/src/main/java/org/eclipse/smarthome/binding/mqttbroker/internal/ssl/PinningSSLContextProvider.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.mqttbroker.internal.ssl;
+
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.naming.ConfigurationException;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+
+import org.eclipse.smarthome.io.transport.mqtt.sslcontext.SSLContextProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is an implementation of the {@link SSLContextProvider}. The {@link PinTrustManager}
+ * will be used for the SSLContext. This implementation forces a TLS 1.2 context.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class PinningSSLContextProvider implements SSLContextProvider {
+    private final Logger logger = LoggerFactory.getLogger(PinningSSLContextProvider.class);
+    final PinTrustManager trustManager;
+
+    public PinningSSLContextProvider(PinTrustManager pinTrustManager) {
+        this.trustManager = pinTrustManager;
+    }
+
+    @Override
+    public SSLContext getContext() throws ConfigurationException {
+        try {
+            SSLContext sslContext = SSLContext.getInstance("TLSv1.2");
```

Comment:
Isn't `"TLSv1.2"`defined as a constant anywhere?

Reply:
Unfortunately not, it's even worse, because Java can decide to return a TLSv1.1 as well if it feels like it. It is more to be understand like, provide me with the minimum of this version an if that is not available, any other recent version. That also means, that we will get TLSv1.3 in the future with the same call.